### PR TITLE
`MediaLibraryContent`: refactor tests to `@testing-library/react`

### PR DESCRIPTION
**Pull request by flootr**
#### Proposed Changes

* `MediaLibraryContent`: refactor tests to `@testing-library/react`

#### Testing Instructions

* Verify unit tests still pass

Related to #63409

### CODE REVIEW

**Review by noahtallen — APPROVED** [review]
Looks great to me :)

File: client/my-sites/media-library/test/content.jsx [hunk 1]
```diff
@@ -20,31 +25,25 @@ const googleConnectionInvalid = {
...
-
-function getMediaContentInstance( props ) {
-	return getMediaContent( props ).instance();
-}
+const defaultProps = {
+	filterRequiresUpgrade: false,
+	mediaScale: 1,
+	mediaValidationErrorTypes: [],
+	selectedItems: [],
+	site: { ID: 1 },
```
Comment (tyxla) [inline client/my-sites/media-library/test/content.jsx]: Wanted to point out that I'm always scared to use `siteId: 1` anywhere, even in tests 😅 Perhaps we could make it a more random number.

File: client/my-sites/media-library/test/content.jsx [hunk 2]
```diff
@@ -20,31 +25,25 @@ const googleConnectionInvalid = {
...
 
 describe( 'MediaLibraryContent', () => {
-	let beforeWindow;
+	let originalScrollTo;
 
-	beforeAll( function () {
-		beforeWindow = global.window;
-		global.window = {};
+	beforeAll( () => {
+		originalScrollTo = window.scrollTo;
```
Comment (tyxla) [inline client/my-sites/media-library/test/content.jsx:41]: 👍 Much better approach than altering the entire `window`.

File: client/my-sites/media-library/test/content.jsx [hunk 3]
```diff
@@ -85,29 +84,29 @@ describe( 'MediaLibraryContent', () => {
 				mediaValidationErrorTypes: [],
 				source: 'google_photos',
 			};
-			const wrapper = getMediaContentInstance();
+			const content = new MediaLibraryContent();
```
Comment (tyxla) [inline client/my-sites/media-library/test/content.jsx:87]: As discussed during pairing the other day, this is an acceptable compromise when we really want to test the methods, given the complexity necessary to setup all needed data to get to that scenario.

**Review by tyxla — APPROVED** [review]
Thanks! 🚀

File: client/my-sites/media-library/test/content.jsx (hunk 1 above)
Comment (flootr) [reply client/my-sites/media-library/test/content.jsx]: Agreed. Adjusted in https://github.com/Automattic/wp-calypso/pull/68668/commits/7e20249ddfad0edabe604fa64b2ad2c90a403bc2

**Pull request merged**